src/ai/backend/agent/kubernetes/intrinsic.py

```diff
@@ -279,8 +332,130 @@ async def extra_info(self) -> Mapping[str, str]:
+            resp = [line for line in raw_resp.split("\n") if "#" not in line and "node_mem" in line]
+            mem_used = resp.pop().split(" ")[1]
+            return mem_used
+
+        async def get_network_stat(core_api, name: str):
+            try:
+                raw_resp = await core_api.connect_get_node_proxy_with_path(
+                    name=name, path="metrics/cadvisor"
+                )
+            except K8sApiException as e:
+                log.warning("kubernetes api error: {}", e)
+                return None
+            net_rx = Decimal(0)
+            net_tx = Decimal(0)
+            q = Decimal(0.000)
+            for line in raw_resp.split("\n"):
+                if "#" in line:
+                    continue
+                if "container_network_receive_bytes" in line:
+                    net_rx += Decimal(line.split(" ")[1]).quantize(q)
+                elif "container_network_transmit_bytes" in line:
+                    net_tx += Decimal(line.split(" ")[1]).quantize(q)
+            return net_rx, net_tx
+
+        await K8sConfig.load_kube_config()
+        core_api = K8sClient.CoreV1Api()
+        nodes = (await core_api.list_node()).to_dict()["items"]
+        node_names = [node["metadata"]["name"] for node in nodes]
+        _mem_tasks = []
+        _net_tasks = []
+        q = Decimal(0.000)
+        for name in node_names:
+            _mem_tasks.append(asyncio.ensure_future(get_memory_stat(core_api, name)))
+            _net_tasks.append(asyncio.ensure_future(get_network_stat(core_api, name)))
+        mem_results = await asyncio.gather(*_mem_tasks)
+        net_results = await asyncio.gather(*_net_tasks)
+        total_mem_used_bytes = Decimal(0)
+        net_rx_bytes = Decimal(0)
+        net_tx_bytes = Decimal(0)
+        for mem_used in mem_results:
+            if mem_used is None:
+                continue
+            total_mem_used_bytes += Decimal(mem_used).quantize(q)
+        for result in net_results:
+            if result is None:
+                continue
+            net_rx_bytes += result[0]
+            net_tx_bytes += result[1]
+        total_mem_capacity_bytes = cast(
+            Decimal,
+            sum(
+                (Decimal(node["status"]["capacity"]["memory"][:-2]) * 1024).quantize(q)
+                for node in nodes
+            ),
+        )
+
+        def get_disk_stat(nodes):
+            total_disk_capacity = Decimal(0)
+            total_disk_usage = Decimal(0)
+            per_disk_stat = {}
+            q = Decimal(0.000)
+            for node in nodes:
+                disk_capacity = node["status"]["capacity"]["ephemeral-storage"]
+                disk_allocatable = node["status"]["allocatable"]["ephemeral-storage"]
+                if disk_capacity.isdigit():
+                    disk_capacity = Decimal(disk_capacity).quantize(q)
+                else:
+                    disk_capacity = (Decimal(disk_capacity[:-2]) * 1024).quantize(q)
```

Comment:
Is this "kilo" conversion guaranteed to be the same in all cases?
Shouldn't we use [`BinarySize`](https://github.com/lablup/backend.ai/blob/0338b2e/src/ai/backend/common/types.py#L417) to parse the scale suffix?